examples/flutter_gallery/lib/demo/material/fab_motion_demo.dart

```diff
@@ -0,0 +1,105 @@
+    FloatingActionButtonPositioner.endFloat, 
+    FloatingActionButtonPositioner.centerFloat,
+    const _TopStartFloatingActionButtonPositioner(),
+  ];
+
+  bool _showFab = true;
+  FloatingActionButtonPositioner _floatingActionButtonPositioner = FloatingActionButtonPositioner.endFloat;
+
+  @override
+  Widget build(BuildContext context) {
+    final Widget floatingActionButton = _showFab 
+      ? new Builder(builder: (BuildContext context) {
+        // We use a widget builder here so that this inner context can find the Scaffold.
+        // This makes it possible to show the snackbar.
+        return new FloatingActionButton(
+          backgroundColor: Colors.yellow.shade900,
+          onPressed: () => _showSnackbar(context),
+          child: const Icon(Icons.add), 
+        );
+      }) 
+      : null;
+    final Widget scaffold = new Scaffold(
+      appBar: new AppBar(title: const Text('FAB Positioner'), bottom: const PreferredSize(preferredSize: const Size.fromHeight(48.0), child: const SizedBox())),
+      floatingActionButtonPositioner: _floatingActionButtonPositioner,
+      floatingActionButton: floatingActionButton,
+      body: new Center(
+        child: new Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            new RaisedButton(
+              onPressed: _moveFab,
+              child: const Text('MOVE FAB'),
+            ),
+            new Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                const Text('Toggle FAB'),
+                new Switch(value: _showFab, onChanged: _toggleFab),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+    return scaffold;
+  }
+
+  void _moveFab() {
+    setState(() {
+      _floatingActionButtonPositioner = _floatingActionButtonPositioners[(_floatingActionButtonPositioners.indexOf(_floatingActionButtonPositioner) + 1) % _floatingActionButtonPositioners.length];
+    });
+  }
+
+  void _toggleFab(bool showFab) {
+    setState(() {
+      _showFab = showFab;
+    });
+  }
+
+  void _showSnackbar(BuildContext context) {
+    Scaffold.of(context).showSnackBar(const SnackBar(content: const Text(_explanatoryText)));
+  }
+}
+
+class _TopStartFloatingActionButtonPositioner extends FloatingActionButtonPositioner {
+  const _TopStartFloatingActionButtonPositioner();
+
+  @override
+  Offset getOffset(ScaffoldPrelayoutGeometry scaffoldGeometry) {
+    double fabX;
+    assert(scaffoldGeometry.textDirection != null);
+    switch (scaffoldGeometry.textDirection) {
+      case TextDirection.ltr:
+        fabX = 16.0 + scaffoldGeometry.horizontalFloatingActionButtonPadding;
+        break;
+      case TextDirection.rtl:
+        fabX = scaffoldGeometry.scaffoldSize.width - scaffoldGeometry.floatingActionButtonSize.width - 16.0 - scaffoldGeometry.horizontalFloatingActionButtonPadding;
```

Comment:
we prefer listing rtl first, see https://master-docs-flutter-io.firebaseapp.com/flutter/intl/TextDirection-class.html

Reply:
Done